docs/user/tutorials/vote-extensions/oracle.md

```diff
@@ -0,0 +1,241 @@
+# Oracle Module leveraging Vote Extensions
+
+## Prerequisites
+
+In this tutorial, we expect the reader to have a chain project already working as we won’t go through the steps of creating a new chain/module.
+
+We also assume you are already familiar with the Cosmos SDK, if you are not we suggest you start with [https://tutorials.cosmos.network](https://tutorials.cosmos.network), as ABCI++ is considered an advanced topic.
+
+## Overview of the project
+
+We’ll go through the creation of a simple price oracle module focusing on the vote extensions implementation, ignoring the details inside the price oracle itself.
+
+We’ll go through the implementation of:
+
+* `ExtendVote` to get information from external price APIs.
+* `VerifyVoteExtension` to check that the format of the provided votes is correct.
+* `PrepareProposal` to process the vote extensions from the previous block and include them into the proposal as a transaction.
+* `ProcessProposal` to check that the first transaction in the proposal is actually a “special tx” that contains the price information.
+* `PreBlocker` to make price information available during FinalizeBlock.
+
+
+## Implement ExtendVote
+
+First we’ll create the `OracleVoteExtension` struct, this is the object that will be marshaled as bytes and signed by the validator.
+
+In our example we’ll use JSON to marshal the vote extension for simplicity but we recommend to find an encoding that produces a smaller output, given that large vote extensions could impact CometBFT’s performance. Custom encodings and compressed bytes can be used out of the box.
+
+```go
+// OracleVoteExtension defines the canonical vote extension structure.
+type OracleVoteExtension struct {
+	Height int64
+	Prices map[string]math.LegacyDec
+}
+```
+
+Then we’ll create a `VoteExtensionsHandler` struct that contains everything we need to query for prices.
+
+```go
+type VoteExtHandler struct {
+	logger          log.Logger
+	currentBlock    int64                            // current block height
+	lastPriceSyncTS time.Time                        // last time we synced prices
+	providerTimeout time.Duration                    // timeout for fetching prices from providers
+	providers       map[string]Provider              // mapping of provider name to provider (e.g. Binance -> BinanceProvider)
+	providerPairs   map[string][]keeper.CurrencyPair // mapping of provider name to supported pairs (e.g. Binance -> [ATOM/USD])
+
+	Keeper keeper.Keeper // keeper of our oracle module
+}
+```
+
+Finally, a function that returns `sdk.ExtendVoteHandler` is needed too, and this is where our vote extension logic will live.
```

Comment:
```suggestion
Finally, a function that returns `sdk.ExtendVoteHandler` is needed, and this is where our vote extension logic will live.
```

maybe a short explanation on what the function does? the tutorial is assuming people need to read the code in order to understand how the oracle works